test/foundry/deploy/03_Erc20Enforcer.t.sol

```diff
@@ -0,0 +1,697 @@
+    holographERC20.safeTransfer(address(erc20Mock), initialValue, zeroBytes);
+  }
+
+  function testSafeTransferFromToErc20() public {
+    mintToDeployer();
+    erc20Mock.toggleWorks(true);
+    vm.expectEmit(true, true, false, true);
+    emit Approval(deployer, alice, initialValue);
+    approvalToAlice(initialValue);
+    //check allowance alice = 1
+    assertEq(holographERC20.allowance(deployer, alice), initialValue);
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), address(erc20Mock), initialValue);
+    vm.prank(alice);
+    holographERC20.safeTransferFrom(address(deployer), address(erc20Mock), initialValue);
+    //check allowance alice = 0
+    assertEq(holographERC20.allowance(deployer, alice), 0);
+  }
+
+  function testSafeTransferFromBytesToErc20() public {
+    mintToDeployer();
+    erc20Mock.toggleWorks(true);
+    vm.expectEmit(true, true, false, true);
+    emit Approval(deployer, alice, initialValue);
+    approvalToAlice(initialValue);
+    //check allowance alice = 1
+    assertEq(holographERC20.allowance(deployer, alice), initialValue);
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), address(erc20Mock), initialValue);
+    vm.prank(alice);
+    holographERC20.safeTransferFrom(address(deployer), address(erc20Mock), initialValue, zeroBytes);
+    //check allowance alice = 0
+    assertEq(holographERC20.allowance(deployer, alice), 0);
+  }
+
+  /*
+   * ERC20 TEST
+   *    Burneable
+   */
+
+  function testBurneableExceedsBalanceRevert() public {
+    vm.expectRevert("ERC20: amount exceeds balance");
+    holographERC20.burn(initialValue);
+  }
+
+  function testBurn() public {
+    mintToDeployer();
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), zeroAddress, initialValue);
+    vm.prank(deployer);
+    holographERC20.burn(initialValue);
+  }
+
+  function testBurnFromNotApproveRevert() public {
+    vm.expectRevert("ERC20: amount exceeds allowance");
+    holographERC20.burnFrom(deployer, initialValue);
+  }
+
+  function testBurnFrom() public {
+    mintToDeployer();
+    approvalToAlice(initialValue);
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), zeroAddress, initialValue);
+    vm.prank(alice);
+    holographERC20.burnFrom(deployer, initialValue);
+  }
+
+  /*
+   * ERC20 TEST
+   *    Permit
+   */
+
+  //TODO buildDomainSeperator
+
+  function testPermitZeroNounce() public {
+    assertEq(holographERC20.nonces(alice), 0);
+  }
+
+  function testPermitBadDeadLineRevert() public {
+    vm.expectRevert("ERC20: expired deadline");
+    holographERC20.permit(deployer, alice, initialValue, badDeadLine, uint8(0x00), zeroSignature, zeroSignature);
+  }
+
+  function testPermitEmptySignatureRevert() public {
+    console.log("block.timestamp", block.timestamp);
+    console.log("block.goodDeadLine", goodDeadLine);
+    console.log("block.badDeadLine", badDeadLine);
```

Comment:
question: why do we need these console.logs? can't we remove them?

Reply:
done!